java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -451,6 +451,22 @@ public static ColumnVector makeList(long rows, DType type, ColumnView... columns
     return new ColumnVector(makeList(handles, type.typeId.nativeId, type.getScale(), rows));
   }
 
+  /**
+   * Create a LIST column from the current column and a given offsets column. The output colum will
+   * contain lists that have elements taken from the current column and their sizes are determined
+   * by the given offsets.
+   *
+   * Note that the users are responsible to make sure the given offsets column is of type INT32 and
+   * it contains valid values to create a LIST column.
```

Comment:
nit: could you clarify this a bit.  The JNI code actually checks to see if it is of type INT32, but from my understanding there are no range checks on the offsets column.  This is a foot gun API because if the user messes up the offsets we start to get bad memory accesses and/or data corruption. It really would be good to have warning to that effect.  It also would be good to say that there is no way to pass in validity.

Reply:
Thanks for this. How to emit warning in Java code?

Reply:
I don't mean logging a wanting. I mean more of adding it to the comments